### PR TITLE
Pre defined sn ps

### PR DESCRIPTION
**Pull request by nick-pestell**
This PR continues development of SNP validation pipeline. For the time being Snippy generated VCF file path is hardcoded in validator.py. 

Seperate functions for simuG with random SNPs and SNPs from snippy vcf. Currently the calling of each respective function is switched on/off via commenting out.

### CODE REVIEW

File: sim/validator.py [hunk 1]
```diff
@@ -30,14 +30,25 @@ def run(cmd, *args, **kwargs):
...
+            "-seed", str(seed)
+        ])
+    else:
+        run([
+            "simuG.pl",
+            "-refseq", reference_path,
+            "-snp_count", str(num_snps),
+            "-prefix", output_path + "simulated",
+            "-seed", str(seed)
+        ])
```
Comment (afishman) [inline sim/validator.py]: This is a bit weird. `predef_snp_path` is a required argument, not possible for the `else` to be run unless you supply it `None` or an empty string

I think a better way to do this is have separate functions. They could be named something along the lines of `simulate_genome_from_vcf` and `simulate_genome_with_random_snps`

Comment (afishman) [reply sim/validator.py]: You can also DRY up out the common parts of the command with a list of strings

File: sim/validator.py [hunk 2]
```diff
@@ -169,13 +186,15 @@ def main():
         description="Performance test btb-seq code")
     parser.add_argument("btb_seq", help="path to btb-seq code")
     parser.add_argument("results", help="path to performance test results")
-    parser.add_argument("--ref", "-r", help="path to reference fasta", default=DEFAULT_REFERENCE_PATH)
-    parser.add_argument("--branch", help="path to reference fasta", default=None)
+    parser.add_argument("--branch", help="name of btb-seq branch to use", default=None)
+    parser.add_argument("--predef_snps", help="optional path to VCF file with predefined SNPs", default=None)
+    parser.add_argument("--num_snps", help="number of simulated snps, overridden by predef_snps", default=16000)
```
Comment (afishman) [inline sim/validator.py]: I don't think we need command line arguments for this. I'm about to add functionality to run multiple samples.

**Review by afishman — CHANGES_REQUESTED** [review]
Looks good, just a few small structural things

File: sim/validator.py (hunk 2 above)
Comment (nick-pestell) [reply sim/validator.py]: Oki doki. How do you think we should define whether to use random or predefined SNPs. And which snippy VCF to use?

Comment (afishman) [reply sim/validator.py]: Leave that for the next PR I think, adding support for multiple samples today, will need to integrate with that. If we have functions for doing both, should  be easy to sort that out.

Comment (nick-pestell) [reply sim/validator.py]: gotcha

File: sim/validator.py [hunk 3]
```diff
@@ -30,25 +30,36 @@ def run(cmd, *args, **kwargs):
...
+            seed (int): Seed value for simulation
+
+        Returns:
+            None
+    """
+    cmd.extend(["-snp_count", str(num_snps),
+                "-seed", str(seed)])
+    run(cmd)
+
+def simulate_genome_from_vcf(cmd, predef_snp_path, seed=1):
```
Comment (afishman) [inline sim/validator.py]: it's inconvenient for users of this code to have to recall the bare-bones command to pass into these functions.

I'd instead write a third function called `simulate_genome` that `simulate_genome_random_snps` and `simulate_genome_from_vcf` call.

Comment (nick-pestell) [reply sim/validator.py]: agreed

**Review by afishman — APPROVED** [review]
saucy

**Pull request merged**